Support for simple begin and end times

ewilligers opened the pull request:
Times may be specified using attributes, or using
beginElement/endElement DOM calls that are exposed by the
SVGAnimationElement IDL.

We don't yet support modifications to begin/end attributes after the element is constructed.

We don't yet support references to other animation begin/end times.

Automated tests still to be added.

ericwilligers committed:
Support for simple begin and end times

Times may be specified using attributes, or using
beginElement/endElement DOM calls that are exposed by the
SVGAnimationElement IDL.

Automated tests stil to be added.

alancutter commented on test/testcases/begin-end-element.html:
```diff
@@ -0,0 +1,51 @@
+        nativeSecondAnim.endElementAt(1);
+        polyfillFirstAnim.endElementAt(2);
+        nativeFirstAnim.endElementAt(2);
+      }, 3500);
```
Add FIXME to use the test harness here.

alancutter commented on test/testcases/freeze-check.js:
```diff
@@ -8,10 +8,10 @@ timing_test(function() {
   at(1500, 'x', 175, polyfillRect, nativeRect);
   at(2000, 'height', 200, polyfillRect, nativeRect);
-  // at(2500, 'width', 125, polyfillRect, nativeRect);
+  at(2500, 'width', 125, polyfillRect, nativeRect);
```
This test doesn't cover multiple values for begin and end.

alancutter commented on smil-in-javascript.js:
```diff
@@ -339,12 +518,91 @@ function createMotionPathAnimation(animationRecord) {
+  // on the element won't be visible to user JavaScript.
+  this.animationRecordId = animationRecordCounter.toString();
+  element.animationRecordId = this.animationRecordId;
+  animationRecords[this.animationRecordId] = this;
```
Move to constructor caller.

alancutter commented:
This change doesn't support modifications to begin/end or references to other animation begin/end times. This should be mentioned in the description.

alancutter commented on smil-in-javascript.js:
```diff
@@ -58,11 +58,97 @@ var observedAttributes = {
+          scheduledAnimationRecords.extractFirst(currentTime))) {
+    animationRecord.processNow();
+  }
+}
```
These can live on a singleton object: {queue: new Queue(), method: function(){this.queue.method()}};

alancutter commented on smil-in-javascript.js:
```diff
@@ -96,6 +182,97 @@ function parseClockValue(value) {
+// Converts value to milliseconds.
+function parseOffsetValue(value) {
+  value = value.trim();
+  if (value.substring(0, 1) === '+') {
```
value[0]

ericwilligers committed:
Retire createAnimationRecord

The logic in createAnimationRecord has been moved to the AnimationRecord
constructor.

Add example with list of begin and end times
For now, simple-list-begin-end.html is a manual test.

alancutter commented:
lgtm
Splitting this file up would make reviewing easier in future.
